src/components/images/imageLoader.js

```diff
@@ -11,22 +12,70 @@ import 'css!./style';
         fillImageElement(elem, source);
     }
 
+    async function itemBlurhashing(target) {
+        let blurhashstr = target.getAttribute('data-blurhash');
+        if (blurhashstr && blurhash.isBlurhashValid(blurhashstr)) {
+            // Although the default values provided by blurhash devs is 32x32, 18x18 seems to be the sweetest spot possible,
+            // cramping up a lot the performance and reducing the memory usage, while retaining almost full blur quality.
+            // Lower values had more visible pixelation
+            let width = 18;
+            let height = 18;
+            let pixels;
+            try {
+                pixels = blurhash.decode(blurhashstr, width, height);
+            } catch (err) {
+                console.error('Blurhash decode error: ', err);
+                return;
+            }
+            let canvas = document.createElement('canvas');
+            canvas.width = width;
+            canvas.height = height;
+            let ctx = canvas.getContext('2d');
+            let imgData = ctx.createImageData(width, height);
+
+            imgData.data.set(pixels);
+            ctx.putImageData(imgData, 0, 0);
+
+            let child = target.appendChild(canvas);
+            child.classList.add('blurhash-canvas');
+            child.style.opacity = 1;
+            if (userSettings.enableFastFadein()) {
+                child.classList.add('lazy-blurhash-fadein-fast');
+            } else {
+                child.classList.add('lazy-blurhash-fadein');
+            }
+
+            target.classList.add('blurhashed');
+        }
+    }
+
+    function switchCanvas(elem) {
+        let child = elem.getElementsByClassName('blurhash-canvas')[0];
+        if (child) {
+            child.style.opacity = elem.getAttribute('data-src') ? 1 : 0;
+        }
+    }
+
     export function fillImage(entry) {
         if (!entry) {
             throw new Error('entry cannot be null');
         }
-
+        let target = entry.target;
         var source = undefined;
-        if (entry.target) {
-            source = entry.target.getAttribute('data-src');
+
+        if (target) {
+            source = target.getAttribute('data-src');
         } else {
             source = entry;
         }
 
+        if (!target.classList.contains('blurhashed') && userSettings.enableBlurhash()) {
+            itemBlurhashing(target);
```

Comment:
```suggestion
            itemBlurhashing(target);
        }
```
you seem to be missing the closing bracket here